src/main/java/BotChat/TaskList.java

```diff
@@ -0,0 +1,272 @@
+     */
+    private Event createEventTask(String description, String from, String to) throws BotChatException {
+        try {
+            return new Event(description, from, to);
+        } catch (Exception e) {
+            throw new BotChatException("Invalid date format. Please use yyyy-MM-dd or yyyy-MM-dd HHmm "
+                    + "format for the event.");
+        }
+    }
+
+    /**
+     * Adds a deadline task to the task list based on the user input.
+     *
+     * @param input The user input containing the deadline task details.
+     * @return A response message or status.
+     * @throws BotChatException If there's an error during the addition process.
+     */
+    public String addDeadlineTask(String input) throws BotChatException {
+        String[] parts = input.split("/", 2);
+        if (parts.length == 2) {
+            String description = parts[0].substring(8);
+            String by = parts[1].substring(3);
+
+            if (!description.isEmpty()) {
+                Deadline deadlineTask = createDeadlineTask(description, by);
+                return addTask(deadlineTask);
+            } else {
+                throw new BotChatException("Please provide a valid description of the task.");
+            }
+        } else {
+            throw new BotChatException("Invalid format of Deadline task. Please try again with the correct format.\n"
+                    + "deadline (event name) /by (deadline)");
+        }
+    }
+
+    /**
+     * Creates a deadline task based on the provided details.
+     *
+     * @param description The description of the deadline task.
+     * @param by          The deadline date/time.
+     * @return The created Deadline task.
+     * @throws BotChatException If there's an error during the creation process.
+     */
+    private Deadline createDeadlineTask(String description, String by) throws BotChatException {
+        if (isValidDateFormat(by)) {
+            return new Deadline(description, by);
+        } else {
+            throw new BotChatException("Invalid date format. Please use yyyy-mm-dd or yyyy-mm-dd HHmm "
+                    + "format for the deadline.");
+        }
+    }
+
+    /**
+     * Checks if the provided date string has a valid date format.
+     *
+     * @param by The date string to be checked.
+     * @return True if the date format is valid, false otherwise.
+     */
+    private boolean isValidDateFormat(String by) {
+        try {
+            LocalDate.parse(by, DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+            return true;
+        } catch (DateTimeParseException e) {
+            try {
+                LocalDateTime.parse(by, DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
+                return true;
+            } catch (DateTimeParseException ex) {
+                return false;
+            }
+        }
+    }
+
+    /**
+     * Adds a tod0 task to the task list based on the user input.
```

Comment:
There may be a slight typo here where "todo" is typed as "tod0", it can be hard to distinguish sometimes 😰.